### PR TITLE
Proposed solution for Issue #82

### PR DESCRIPTION
**Pull request by vin-fandemand**
This is a proposed solution for 
[https://github.com/dart-lang/tools/issues/345](url)

### CODE REVIEW

**Comment by vin-fandemand** [comment]
@googlebot I signed it!

**Comment by vin-fandemand** [comment]
Related to https://github.com/dart-lang/pub/pull/2588

**Comment by hayshi** [comment]
@googlebot I signed it!

File: lib/src/authorization_code_grant.dart [hunk 1]
```diff
@@ -225,8 +225,8 @@ class AuthorizationCodeGrant {
   /// [FormatError] if the `state` parameter doesn't match the original value.
   ///
   /// Throws [AuthorizationException] if the authorization fails.
-  Future<Client> handleAuthorizationResponse(
-      Map<String, String> parameters) async {
+  Future<Client> handleAuthorizationResponse(Map<String, String> parameters,
+      {bool useIdToken = false}) async {
```
Comment (jonasfj) [inline lib/src/authorization_code_grant.dart:229]: I agree that there are use-cases for using `id_token`, but in general that's not what `id_token` is for.. and in oauth the `access_token` is the token applications should use for accessing the APIs that they have been granted access to using the oauth2 flow. Is that not correct?

It feels a bit like we're just adding extra functionality that might be better placed in an ODIC specific package.
See suggestion in https://github.com/dart-lang/oauth2/issues/82#issuecomment-750264163

TL;DR: I'm a bit concerned this is outside the scope of this package, and that increasing the scope of this package will increase bugs significantly -- and we're already somewhat slow to respond to issues here. So putting ODIC logic in a separate package whether a fork or not, might be a better option.

Comment (eirikskogoy) [reply lib/src/authorization_code_grant.dart:229]: Would really like this PR to be implemented to be able to use the idToken as the authorization. Is this still of the table to do?

**Comment by mosuem** [comment]
As this repo has moved to https://github.com/dart-lang/tools/tree/main/pkgs/oauth2, please refile the PR there. Thanks!
